### PR TITLE
Interactive To-Do List 📝✅

### PR DESCRIPTION
**Pull request by DevKaranJ**
Create an engaging to-do list application with task completion features. This project highlights Webpack for JavaScript bundling, ES6 syntax for modern coding, and ES6 modules for organized development. Users can easily check off tasks, revert completions, clear done tasks, and keep their list intact even after page reloads via local storage.

Features:

- 📝 Task list with checkboxes for completion status.
- ✅ Toggle tasks as done or undone via checkboxes.
- 🗑️ Quickly remove completed tasks with a dedicated button.
- 💾 Store task list in local storage for persistent use.

Documentation:

- 📚 Clear code comments to enhance understanding.
- 📖 Concise README offering setup guidance and a project snapshot.

### CODE REVIEW

File: src/modules/checkboxes.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    completedTasks.forEach((task) => {
+      const listItem = findParentListItem(task);
+      if (listItem) {
+        listItem.remove();
+      }
+    });
+
+    saveTasksToLocalStorage(taskList);
+  });
+}
```
Comment (Gambit142) [inline src/modules/checkboxes.js]: - [x] Great work implementing the clear all completed function. However, based on the requirements, you ought to use the JavaScript `filter()` method to remove all completed tasks. Kindly use this method. 👍 😎

File: src/modules/checkboxes.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      const listItem = findParentListItem(event.target);
+      if (listItem) {
+        const taskText = listItem.querySelector('.task-item span'); // Updated selector
+        taskText.classList.toggle('completed', event.target.checked);
+
+        saveTasksToLocalStorage(taskList);
+      }
+    }
+  });
+}
```
Comment (Gambit142) [inline src/modules/checkboxes.js]: - [x] The status update functionality should toggle the `completed` key of every task object between `true` and `false`. Currently, each of your task's objects doesn't have a `completed` key. I would advise you to include it when creating a task. Then when a user clicks on the checkbox for each task, it should update the `completed` key to `true` or `false`. Kindly implement this change. 👍 😎 

image

File: src/modules/taskListFunctions.js [hunk 3]
```diff
@@ -3,11 +3,12 @@ export function addTaskToList(taskList, taskText, index) {
   const li = document.createElement('li');
   const newIndex = taskList.children.length + 1;
   li.innerHTML = `
```
Comment (Gambit142) [inline src/modules/taskListFunctions.js:5]: - [OPTIONAL] This change is related to the styling of your application. The delete button for the application is outside the parent container. Kindly style it properly. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @DevKaranJ 👋, 

Good job so far!

## Highlights
- Setup linters properly ✔️ 
- Created a new JavaScript file for new functionality ✔️ 
- README is professional ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/modules/taskFunctions.js [hunk 4]
```diff
@@ -2,9 +2,10 @@
 export function saveTasksToLocalStorage(taskList) {
   const tasks = [];
   const taskItems = taskList.querySelectorAll('li');
-  taskItems.forEach((taskItem, index) => {
-    const text = taskItem.querySelector('span').textContent;
-    tasks.push({ index: index + 1, text });
+  taskItems.forEach((taskItem) => {
+    const text = taskItem.querySelector('.task-item span').textContent;
+    const completed = taskItem.querySelector('.task-checkbox').checked;
+    tasks.push({ text, completed });
```
Comment (cynthiainga) [inline src/modules/taskFunctions.js]: - [x] Good job implementing requested changes but now tasks don't have indexes, each task should have an index and this is part of the project but also make sure that after deleting tasks the index should be updated also for the remaining tasks, Please fix this 🙏🏾 

Screenshot 2023-08-26 at 17 19 27

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @DevKaranJ, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights

- [x] No linters errors ✔️ 
- [x] Descriptive PR ✔️ 
- [ ] Created a new JavaScript file for new functionality ✔️

 ## Required Changes ♻️

- [x] Please kindly check the comment under the review 👇🏾 👇🏾 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @DevKaranJ      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional Readme
- [x] Interactive list functioning  as intended 💯 
- [x] Requested changes implemented. 



 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**